rocprim/include/rocprim/device/detail/device_binary_search.hpp

```diff
@@ -46,7 +46,7 @@ Size lower_bound_n(RandomAccessIterator first,
     while(left < right)
     {
         const Size mid = get_binary_search_middle(left, right);
-        if(compare_op(first[mid], value))
+        if(compare_op(typename std::iterator_traits<RandomAccessIterator>::value_type(first[mid]), value))
```

Comment:
We're working on this, I think this is a thrust-related issue and the fix is needed in the default comparator we use in rocThrust.
@nolmoonen currently writing some tests and looking in further

Reply:
This is indeed the case, and requires a fix in rocThrust instead of rocPRIM. I have opened a PR here: https://github.com/ROCmSoftwarePlatform/rocThrust/pull/263.